### PR TITLE
Add missing ifdefs for Harmony artifacts

### PR DESCRIPTION
**Pull request by pdbain-ibm**
This is part of https://github.com/eclipse/openj9/issues/529

Signed-off-by: Peter Bain <peter_bain@ca.ibm.com>

### CODE REVIEW

**Comment by pdbain-ibm** [comment]
I test compiled the OpenJ9 JVM on macOS and the IBM Linux and Windows JVMs on the internal build farm.

File: runtime/ddr/ddrtable.c [hunk 1]
```diff
@@ -33,7 +33,9 @@ extern const J9DDRStructDefinition *getJITAutomatedStructTable(struct OMRPortLib
 extern const J9DDRStructDefinition *getOmrStructTable(struct OMRPortLibrary *portLib);
 extern const J9DDRStructDefinition *getVMStructTable(struct OMRPortLibrary *portLib);
 extern const J9DDRStructDefinition *getStackWalkerStructTable(struct OMRPortLibrary *portLib);
+#ifdef J9VM_OPT_HARMONY
```
Comment (keithc-ca) [inline runtime/ddr/ddrtable.c]: Please use `#if defined(J9VM_OPT_HARMONY)` instead.

Comment (pdbain-ibm) [reply runtime/ddr/ddrtable.c]: Done.  I did not change the existing `#ifdef`s, since we hope they will go away soon-ish.

**Comment by keithc-ca** [comment]
Jenkins test sanity,extended plinux jdk8,jdk11

**Comment by keithc-ca** [comment]
Jenkins test sanity,extended plinux jdk8,jdk11

**Comment by keithc-ca** [comment]
Jenkins test sanity plinux jdk11

**Pull request merged**